lib/faster_s3_url/builder.rb

```diff
@@ -223,6 +232,28 @@ def uri_escape_key(string)
       end
     end
 
+    # Handle endpoint, modifying host or path with bucketname, and setting
+    # host.
+    #
+    # if none set, set default host.
+    #
+    # Set base_url correct for host or endpoint.
+    def parsed_base_uri(bucket_name:, host:, endpoint:)
```

Comment:
This is more of a style suggestion, as I like to offer alternative options during code reviews.

I think the `return` calls here are extra since ruby uses the return value of `if..else`. I generally only use return when it’s specifically needed.

This might be a great method to use a few guard clauses to catch edge cases early, and limit the nesting.

```ruby
    def parsed_base_uri(bucket_name:, host:, endpoint:)
      return URI.parse("https://#{host}") if host
      return URI.parse("https://#{default_host(bucket_name)}") if endpoint.nil?
      
      parsed = URI.parse(endpoint)
      if parsed.host =~ /\A\d+\.\d+\.\d+\.\d+\Z/
        parsed.path = "/#{bucket_name}"
      else
        parsed.host = "#{bucket_name}.#{parsed.host}"
      end
      parsed
    end
```

Reply:
I like your version, much more readable, thanks. I meant to come back and refactor this one for readability but forgot!